src/k8s/pkg/k8sd/setup/k8s_apiserver_proxy_test.go

```diff
@@ -130,4 +130,29 @@ func TestK8sApiServerProxy(t *testing.T) {
 		// Compare the expected endpoints with those in the file
 		g.Expect(config.Endpoints).To(Equal(endpoints))
 	})
+
+	t.Run("IPv6", func(t *testing.T) {
+		g := NewWithT(t)
+
+		// Create a mock snap
+		s := mustSetupSnapAndDirectories(t, setKubeletMock)
+		s.Mock.Hostname = "dev"
+
+		g.Expect(setup.K8sAPIServerProxy(s, nil, "[2001:db8::]", nil)).To(Succeed())
```

Comment:
> drive-by fix: brackets around the IPv6 are only required/accepted if combined with a port.

Isn't `[2001:db8::]` in conflict with the above statement?

Reply:
Same as above - in the implementation we add a port.